providers/aws/aws_provider.py

```diff
@@ -20,20 +21,19 @@ class AWS_Credentials:
 @dataclass
 class Input_Data:
     profile: str
-    role_name: str
-    account_to_assume: str
+    role_arn: str
```

Comment:
I would merge this three data models into two, one big object with the input credentials and the session, each in their own object.

Also, I would also include the original credentials in this object not to play with two different objects. For me using OOP is easier to access something like this `session.original_session` and `session.current_session`.